nixos/modules/services/networking/wireguard.nix

```diff
@@ -182,7 +186,8 @@ let
   wgCommand = "${pkgs.wireguard}/bin/wg";
 
   generateUnit = name: values:
-    assert (values.privateKey != null || values.privateKeyFile != null);
+    # exactly one way to specify the private key must be set
+    assert (values.privateKey != null) != (values.privateKeyFile != null);
```

Comment:
This is good. But you need something similar for preshared keys: only one of the keys may be specified, but not both.